game/unicode.cc

```diff
@@ -10,16 +10,40 @@
 #include <unicode/ubidi.h>
 #include "compact_enc_det/compact_enc_det.h"
 
-icu::ErrorCode UnicodeUtil::m_staticIcuError = icu::ErrorCode();
-icu::RuleBasedCollator UnicodeUtil::m_dummyCollator (icu::UnicodeString (""), icu::Collator::PRIMARY, m_staticIcuError);
-icu::RuleBasedCollator UnicodeUtil::m_sortCollator  (nullptr, icu::Collator::SECONDARY, m_staticIcuError);
+std::unique_ptr<icu::RuleBasedCollator> UnicodeUtil::m_searchCollator = nullptr;
+std::unique_ptr<icu::RuleBasedCollator> UnicodeUtil::m_sortCollator = nullptr;
```

Comment:
Initialization not needed. Smartpointers are by default Null.

Reply:
It's because they're static.

Reply:
Sure, but the `= nullptr` is not needed ;-)